### PR TITLE
Refactor error handling

### PR DESCRIPTION
**Pull request by dmerejkowsky**

### CODE REVIEW

File: src/python_info.rs [hunk 1]
```diff
@@ -49,5 +56,9 @@ fn get_python_binary(requested_python: &Option<String>) -> Result<std::path::Pat
     }
 
     // Python3 may be called 'python', for instance on Windows
-    Ok(which::which("python")?)
+    let res = which::which("python");
+    if res.is_err() {
+        return error::new("Neither `python3` nor `python` fonud in PATH");
+    }
+    Ok(res.unwrap())
```
Comment (tux3) [inline src/python_info.rs]: Same idea with `map_err`, that's maybe just me but I find it a little more idiomatic!

```rust
which::which("python").map_err(|e| Error::Other { 
    message: "Neither `python3` nor `python` fonud in PATH".to_string() 
})
```

File: tests/helpers/mod.rs [hunk 2]
```diff
@@ -24,15 +24,18 @@ impl TestApp {
...
             include_str!("../../demo/test_demo.py"),
-        ).expect("");
+        )
+        .expect("");
         std::fs::write(
             self.tmp_path.join("setup.py"),
             include_str!("../../demo/setup.py"),
-        ).expect("");
+        )
+        .expect("");
```
Comment (tux3) [inline tests/helpers/mod.rs]: If you're expecting with an empty string, shouldn't you just `unwrap()` ?

File: src/python_info.rs (hunk 1 above)
Comment (dmerejkowsky) [reply src/python_info.rs]: Agree. Let's try that

File: src/venv_manager.rs [hunk 3]
```diff
@@ -172,21 +173,22 @@ impl VenvManager {
     fn create_venv(&self) -> Result<(), Error> {
         let parent_venv_path = &self.paths.venv.parent();
         if parent_venv_path.is_none() {
-            return error::new("venv_path has no parent");
+            return Err(Error::Other {
+                message: "venv_path has no parent".to_string(),
+            });
         }
         let parent_venv_path = parent_venv_path.unwrap();
```
Comment (tux3) [inline src/venv_manager.rs]: You can also use combinators on Options if you want, not just Results!
The 7 lines above can be:

```rust
let parent_venv_path = &self.paths.venv.parent().ok_or(Error::Other {
    message: "venv_path has no parent".to_string(),
})?;
```

**Pull request merged**